calico_rkt/calico_rkt.py

```diff
@@ -1,209 +1,227 @@
+                      orchestrator_id=ORCHESTRATOR_ID,
+                      container_id=container_id)
 
-    interface = env['CNI_IFNAME']
+    # Delete profile if only member
+    if datastore_client.profile_exists(net_name) and \
+       len(datastore_client.get_profile_members(net_name)) < 1:
+        try:
+            _log.info("Profile %s has no members, removing from datastore" % net_name)
+            datastore_client.remove_profile(net_name)
+        except:
+            _log.error("Cannot remove profile %s: Profile cannot be found." % container_id)
+            sys.exit(1)
 
-    endpoint, ip = _container_add(hostname=HOSTNAME,
-                                orchestrator_id=ORCHESTRATOR_ID,
-                                container_id=container_id,
-                                netns_path=netns_path,
-                                interface=interface,
-                                client=client)
-    print_stderr('Finished configuring network interface', file=sys.stderr)
-    return endpoint, ip
 
-def _container_add(hostname, orchestrator_id, container_id, netns_path, interface, client):
+def _create_calico_endpoint(container_id, netns_path, interface, subnet):
     """
-    Add a container to Calico networking with the given IP
-    Return Endpoint object
+    Configure the Calico interface for a pod.
+    Return Endpoint and IP
     """
+    _log.info('Configuring Calico networking.')
+
     try:
-        _ = client.get_endpoint(
-            hostname=hostname,
-            orchestrator_id=orchestrator_id,
-            workload_id=container_id
-        )
+        _ = datastore_client.get_endpoint(hostname=HOSTNAME,
+                                          orchestrator_id=ORCHESTRATOR_ID,
+                                          workload_id=container_id)
     except KeyError:
         # Calico doesn't know about this container.  Continue.
         pass
     else:
-        print_stderr("This container has already been configured with Calico Networking.")
+        _log.error("This container has already been configured with Calico Networking.")
         sys.exit(1)
 
-    # Allocate and Assign ip address through IPAM Client
-    ip = _allocate_IP()
+    endpoint = _container_add(hostname=HOSTNAME,
+                                  orchestrator_id=ORCHESTRATOR_ID,
+                                  container_id=container_id,
+                                  netns_path=netns_path,
+                                  interface=interface,
+                                  subnet=subnet)
+
+    _log.info('Finished configuring network interface')
+    return endpoint
+
+
+def _container_add(hostname, orchestrator_id, container_id, netns_path, interface, subnet):
+    """
+    Add a container to Calico networking
+    Return Endpoint object and newly allocated IP
+    """
+    # Allocate and Assign ip address through datastore_client
+    pool, ip = _assign_to_pool(subnet)
```

Comment:
I'm still not convinced this needs its own function, I think you can perform it inline here. but I'm 50/50